pkg/reconciler/ingress/ingress.go

```diff
@@ -487,9 +374,5 @@ func getLBStatus(gatewayServiceURL string) []v1alpha1.LoadBalancerIngressStatus
 func (r *Reconciler) shouldReconcileTLS(ing *v1alpha1.Ingress) bool {
```

Comment:
Instead of leaving the HTTPS servers in the Gateway when `IngressTLS` is removed, could we always try to reconcile `IngressTLS` no matter if `IngressTLS` is empty or not?
Ideally with my change https://github.com/knative/serving/pull/6937, IngressTLS reconcilation should be idempotent. Reconciling an empty IngressTLS will try to delete the leftover HTTPS servers belonging to the original Ingress.
WDYT?

Reply:
I like the idea.